app/src/main/java/com/asfoundation/wallet/onboarding_new_payment/amazonPay/OnboardingAmazonPayFragment.kt

```diff
@@ -0,0 +1,334 @@
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.fragment.app.viewModels
+import com.appcoins.wallet.core.analytics.analytics.common.ButtonsAnalytics
+import com.appcoins.wallet.core.analytics.analytics.legacy.BillingAnalytics
+import com.appcoins.wallet.core.utils.android_common.CurrencyFormatUtils
+import com.appcoins.wallet.core.utils.properties.HostProperties
+import com.appcoins.wallet.ui.common.theme.WalletColors.styleguide_blue_secondary
+import com.appcoins.wallet.ui.common.theme.WalletColors.styleguide_dark_grey
+import com.appcoins.wallet.ui.common.theme.WalletColors.styleguide_pink
+import com.appcoins.wallet.ui.common.theme.WalletColors.styleguide_white
+import com.appcoins.wallet.ui.widgets.GenericError
+import com.appcoins.wallet.ui.widgets.component.Animation
+import com.appcoins.wallet.ui.widgets.component.ButtonType
+import com.appcoins.wallet.ui.widgets.component.ButtonWithText
+import com.asf.wallet.R
+import com.asfoundation.wallet.billing.amazonPay.AmazonPayIABFragment
+import com.asfoundation.wallet.billing.amazonPay.models.AmazonConst.Companion.APP_LINK_HOST
+import com.asfoundation.wallet.billing.amazonPay.models.AmazonConst.Companion.APP_LINK_PATH
+
+import com.asfoundation.wallet.billing.amazonPay.models.AmazonConst.Companion.createAmazonTransactionLink
+import com.asfoundation.wallet.onboarding_new_payment.OnboardingPaymentEvents
+import com.asfoundation.wallet.onboarding_new_payment.getPurchaseBonusMessage
+import com.asfoundation.wallet.ui.iab.WebViewActivity
+import com.wallet.appcoins.core.legacy_base.BasePageViewFragment
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+
+@AndroidEntryPoint
+class OnboardingAmazonPayFragment : BasePageViewFragment() {
+
+  private val viewModel: OnboardingAmazonPayViewModel by viewModels()
+  lateinit var args: OnboardingAmazonPayFragmentArgs
+
+  @Inject
+  lateinit var navigator: OnboardingAmazonPayNavigator
+
+  @Inject
+  lateinit var analytics: OnboardingPaymentEvents
+
+  @Inject
+  lateinit var formatter: CurrencyFormatUtils
+
+
+  @Inject
+  lateinit var buttonsAnalytics: ButtonsAnalytics
+  private val fragmentName = this::class.java.simpleName
+
+  private lateinit var resultAuthLauncher: ActivityResultLauncher<Intent>
+
+  override fun onCreateView(
+    inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+  ): View {
+    args = OnboardingAmazonPayFragmentArgs.fromBundle(requireArguments())
```

Comment:
why use this way of getting the args?

its easier just to initialize it like this:

`private val args by navArgs<OnboardingAmazonPayFragmentArgs>()`